runtime/doc/lsp.txt

```diff
@@ -454,30 +454,44 @@ You probably want these inside a |ColorScheme| autocommand.
 Use |LspTokenUpdate| and |vim.lsp.semantic_tokens.highlight_token()| for more
 complex highlighting.
 
-The following groups are linked by default to standard |group-name|s:
-
-@lsp.type.class          Structure
-@lsp.type.decorator      Function
-@lsp.type.enum           Structure
-@lsp.type.enumMember     Constant
-@lsp.type.event          Type
-@lsp.type.function       Function
-@lsp.type.interface      Structure
-@lsp.type.keyword        Keyword
-@lsp.type.macro          Macro
-@lsp.type.method         Function
-@lsp.type.modifier       Type
-@lsp.type.namespace      Structure
-@lsp.type.number         Number
-@lsp.type.operator       Operator
-@lsp.type.parameter      Identifier
-@lsp.type.property       Identifier
-@lsp.type.regexp         SpecialChar
-@lsp.type.string         String
-@lsp.type.struct         Structure
-@lsp.type.type           Type
-@lsp.type.typeParameter  TypeDef
-@lsp.type.variable       Identifier
+The following is a list of standard captures used in queries for Nvim,
+highlighted according to the current colorscheme (use |:Inspect| on one to see
+the exact definition):
+
+@lsp.type.class          Identifiers that declare or reference a class type
+@lsp.type.comment        Tokens that represent a comment
```

Comment:
To render this list with "flow" layout (as all Nvim-owned help pages must be), would you mind adding a `-` before each list item. See https://neovim.io/doc/user/lsp.html#lsp-api for an example of how a list is rendered.

Without this, the parser can't know if this is intended to be a list or a paragraph (soft-wrapped).

Alternative workaround is to put these in a codeblock.

Reply:
This would break highlighting in Nvim, which I consider higher priority.

Reply:
Not sure why _this_ list breaks, but the equivalent treesitter group list works fine. Need to investigate.

Reply:
> To render this list with "flow" layout (as all Nvim-owned help pages must be), would you mind adding a `-` before each list item. See https://neovim.io/doc/user/lsp.html#lsp-api for an example of how a list is rendered.
> 
> Without this, the parser can't know if this is intended to be a list or a paragraph (soft-wrapped).
> 
> Alternative workaround is to put these in a codeblock.

This follows the [similar documentation for tree-sitter groups](https://github.com/neovim/neovim/blob/9765efb40f57b543d9fea610fb0d0f05b8387454/runtime/doc/treesitter.txt#L468-L585).

Reply:
> This follows the [similar documentation for tree-sitter groups](https://github.com/neovim/neovim/blob/9765efb40f57b543d9fea610fb0d0f05b8387454/runtime/doc/treesitter.txt#L468-L585).

That also needs to be fixed, eventually...

Reply:
No, it doesn't. This is working fine.

Reply:
It's not working fine in "flow" layout. https://neovim.io/doc/user/treesitter.html#treesitter-parsers currently uses hardwrap layout.

Migrating to "flow" layout will eventually allow us to soft-wrap `:help` presentation in Nvim itself.

Reply:
Is there a list somewhere I can see easily which files use this "flow" layout? Consider adding these (external) constraints to the `dev` documentation.

In any case, this is then a global issue and need not block this PR; I need to invest some time to find a solution that works for `:help` (which is the primary goal) and this flow layout. Both codeblocks (probably) and list markers (definitely) will break the former.

Reply:
> Is there a list somewhere I can see easily which files use this "flow" layout?

https://github.com/neovim/neovim/blob/9765efb40f57b543d9fea610fb0d0f05b8387454/scripts/gen_help_html.lua#L58-L73

> Consider adding these (external) constraints to the dev documentation.

it's in `:help dev-doc`: 

   For Nvim-owned docs, use the following strict subset of "vimdoc" to ensure the
   help doc renders nicely in other formats (such as HTML:
   https://neovim.io/doc/user ).

> In any case, this is then a global issue and need not block this PR;

sure, doesn't need to block this PR.

Reply:
That documentation is a bit confusing: are lists auto-wrapped or not?

Reply:
List items are soft-wrapped just like paragraphs/blocks.

Without list items (this PR), the block is treated as a paragraph, because the structure is not recognized. List items communicate the structure.

Reply:
This is a table, though _not_ a list... (So I don't want to wrap these, at all. And treating them as code blocks interferes with the highlighting, which is a big usability feature.)

Reply:
Yes, but tables currently aren't supported. It doesn't (normally) cause any problem with `:help` rendering, so it's a reasonable workaround.

Reply:
If highlighting is somehow broken with a list, why would it work better in a table (which would require some other syntax; it wouldn't be identical to what's currently here).

Reply:
No, that's not the problem here. I can adapt to consistent(!) formatting that doesn't leverage other syntax highlighting elements. (Probably. Under leveled; need to grind some more.)

Reply:
> This would break highlighting in Nvim, which I consider higher priority.

This sounds like a fixable problem, no?

At some point I'd like to transfer these docs into lua code documentation as Markdown, which indeed would require this to be a list (or table) for this not to be wrapped.